Sources/TuistKit/Services/Inspect/InspecImplicitImportsService.swift

```diff
@@ -46,41 +72,74 @@ final class InspectImplicitImportsService {
         self.targetScanner = targetScanner
     }
 
-    func run(path: String?) async throws {
+    func run(
+        path: String?,
+        inspectType: InspectType
+    ) async throws {
         let path = try self.path(path)
         let config = try await configLoader.loadConfig(path: path)
         let generator = generatorFactory.defaultGenerator(config: config, sources: [])
         let graph = try await generator.load(path: path)
-        let issues = try await lint(graphTraverser: GraphTraverser(graph: graph))
+        let issues = try await lint(graphTraverser: GraphTraverser(graph: graph), inspectType: inspectType)
         guard issues.isEmpty else {
-            throw InspectImplicitImportsServiceError.implicitImportsFound(issues)
+            switch inspectType {
+            case .redundant:
+                throw InspectRedundantImportsServiceError.redundantImportsFound(issues)
+            case .implicit:
+                throw InspectImplicitImportsServiceError.implicitImportsFound(issues)
+            }
         }
-        logger.log(level: .info, "We did not find any implicit dependencies in your project.")
+        logger.log(
+            level: .info,
+            "We did not find any \(inspectType == .implicit ? "implicit" : "redundant") dependencies in your project."
+        )
     }
 
-    private func lint(graphTraverser: GraphTraverser) async throws -> [InspectImplicitImportsServiceErrorIssue] {
-        let allTargets = graphTraverser
+    private func lint(
+        graphTraverser: GraphTraverser,
+        inspectType: InspectType
+    ) async throws -> [InspectImportsServiceErrorIssue] {
```

Comment:
I'd move this to a service its own that would be shared between `InspectImplicitImportsService` and `InspectRedundantImportsService`. For example, it can be called `GraphImportsLinter`.

Additionally, I'd return `LintingIssue` instead of a custom issue that is then transformed into an error. We don't need to follow the `GraphLinting` protocol 1-1, but we should be _very_ close to it. They both solve a very similar problem, so there's no need to diverge here.